app/uk/gov/hmrc/securemessage/connectors/EISConnector.scala

```diff
@@ -49,30 +47,29 @@ class EISConnector @Inject()(
   private val eisEndpoint = servicesConfig.getString("microservice.services.eis.endpoint")
   private val eisEnvironment = servicesConfig.getString("microservice.services.eis.environment")
 
-  def forwardMessage(queryResponse: QueryResponseWrapper): Future[Either[EisForwardingError, Unit]] = {
+  def forwardMessage(queryMessageRequest: QueryMessageRequest): Future[Either[EisForwardingError, Unit]] = {
     implicit val hc: HeaderCarrier = HeaderCarrier()
     httpClient
-      .doPut[QueryResponseWrapper](
+      .doPut[QueryMessageRequest](
         s"$eisBaseUrl$eisEndpoint",
-        queryResponse,
+        queryMessageRequest,
         Seq(
           (CONTENT_TYPE, MimeTypes.JSON),
           (ACCEPT, MimeTypes.JSON),
           (AUTHORIZATION, s"Bearer $eisBearerToken"),
           (DATE, DateTime.now().toString("EEE, dd MMM yyyy HH:mm:ss z")),
-          (CustomHeaders.CorrelationId, UUID.randomUUID().toString),
+          (CustomHeaders.CorrelationId, queryMessageRequest.requestCommon.acknowledgementReference),
```

Comment:
do we need request and session ids as well ?

Reply:
no